FoundationsMastery/CustomString.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Linq;
+using System;
+using System.Collections.Generic;
+using FoundationsMastery.Interfaces;
+using System.Collections;
+
+namespace FoundationsMastery
+{
+    public class CustomString : ICustomString
+    {
+        public IEnumerable<char> Contents { get; set; }
+
+        public CustomString(char[] mychars)
+        {
+            Contents = mychars;
+        }
+
+        public CustomString()
+        {
+        }
+
+        public int Length
+        {
+            get
+            {
+                return Contents.Count();
+            }
+        }
+
+        public void Clear()
+        {
+            return myString.Clear();
+        }
+
+        public string Concat(IEnumerable<char> rhs)
+        {
+            return String.Concat(rhs);
```

Comment:
`String.Concat` is a great method to use - however, you need 2 things (strings) to join together. `lhs` and `rhs` are standard ways of referring to the 2 arguments in a binary function (a function with two arguments, e.g. Math.Add(1, 2) => returns 3), and stand for 'left hand side' and 'right hand side'. The way this method is used looks like this:
```
var myString = new CustomString(new char[] { 'a', 'b', 'c' };
myString.Concat("123");       // => returns "abc123"
```
What would the implied "lhs" be in this object?